### PR TITLE
Basics-9-11 Anna Leong Project 3 Blackjack

### PR DESCRIPTION
**Pull request by annaleongmy**
Please fill out the survey before submitting the pull request. Thanks!

🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀

How many hours did you spend on this assignment?
5-6h

Please fill in one error and/or error message you received while working on this assignment.
Within my main function, I realised I had to call the score calculation helper functions within each if statement instead of placing it outside of the if statements in order for it to calculate accurately. However I feel this makes the code a bit repetitive.

What part of the assignment did you spend the most time on?
Figuring out the logic for when the aces should be counted as 1s or 11s was a bit of a mind-bender, but I found this Quora page quite helpful to work it out: https://www.quora.com/When-is-an-ace-considered-as-1-or-11-in-Blackjack 

Comfort Level (1-5):
4

Completeness Level (1-5):
5

What did you think of this deliverable?
The jump between project 2 and 3 was less than between project 1 and 2, so I found it more manageable. The lessons/class exercises were also helpful to push us along. Being able to copy/modify some of the code from class exercises saved me some time.

Is there anything in this code that you feel pleased about?
I found loops rather hard to grasp at first, but I'm happy I managed to use them a few times in this exercise to display the full hands of the players, count the scores and write the logic for when the dealer has to hit. I'm also pleased that at the end of the main function, I managed to find a way to return the cards to the deck to start a new game without having to hit the refresh button.

### CODE REVIEW

**Comment by annaleongmy** [comment]
Please fill out the survey before submitting the pull request. Thanks!

🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀

How many hours did you spend on this assignment?
8h

Please fill in one error and/or error message you received while working on this assignment.
I was only able to figure out how to do the multiplayer function up till hit or stand mode.

What part of the assignment did you spend the most time on?
The multiplayer mode was very challenging for me and I'm really bummed I couldn't figure it out in full ):

Comfort Level (1-5):
3

Completeness Level (1-5):
2

What did you think of this deliverable?
Turnaround time between project 3 part 1 and 2 was too short for me, especially between Tues to Sat as I am working fulltime so only had weeknights to try to complete this

Is there anything in this code that you feel pleased about?
I was able to reference the loop function for making the card deck to write a loop to create a nested array for the player hands in multiplayer mode

File: script.js [hunk 1]
```diff
@@ -1,4 +1,598 @@
...
+8. If you bust, the dealer wins your bet<br>
+9. After all players are done, the dealer will hit if their score is less than 17<br>
+10. If the dealer busts, every player still in the game wins 2 times their bet<br>
+11. If dealer does not bust, players who score higher than the dealer win 2 times their bet<br>
+12. Everyone else loses their bet to the dealer
+<br><br>
+Please enter number of players to start.`
+
+//keeps track of number of players
+var numPlayers = 0;
```
Comment (michelle-mok) [inline script.js:91]: good use of functions to simplify code!

File: script.js [hunk 2]
```diff
@@ -1,4 +1,598 @@
...
+  numPlayers = Number(number);
+  return `There are ${numPlayers} players in this game.`;
+};
+
+//creates player hands
+var createPlayerHands = function() {
+  var playerHands = [];
+  var playerIndex = 1;
+    while (playerIndex <= numPlayers) {
+      var playerHand = {
```
Comment (michelle-mok) [inline script.js:102]: excellent use of objects

File: script.js [hunk 3]
```diff
@@ -1,4 +1,598 @@
...
+  };
+  return `It's player ${playerHands[playerTurn].player}'s turn.`
+};
+
+//creates a player's bet on their turn
+var createBet = function(input,playerTurn){
+  var bet = Number(input);
+  playerHands[playerTurn].bank -= bet;
+  //checks that bet is not blank and is a number
+  if (Number.isNaN(bet) ||
```
Comment (michelle-mok) [inline script.js:139]: if you're using if/return, you don't need to use else as well

```
if(playerHands[playerTurn].bank < 0){
    playerHands[playerTurn].bank +=bet;
    return `Player ${playerHands[playerTurn].player} doesn't have enough in the bank!<br>
    Player ${playerHands[playerTurn].player} bank: ${playerHands[playerTurn].bank}`;
  }

 if (playerHands[playerTurn].player < numPlayers){
    playerHands[playerTurn].bet = bet;
    var playerBetMessage = `Player ${playerHands[playerTurn].player} bet $${playerHands[playerTurn].bet}.`
    var nextTurnMessage = nextTurn();
    return `${playerBetMessage}<br>${nextTurnMessage}` }
```
if you use if /else if, you return at the end of the code block, not at the end of every if statement

**Review by michelle-mok — COMMENTED** [review]
good use of functions to simplify code
excellent work keep it up!
 thanks for all the hard work!

**Pull request closed without merging**